great_expectations/checkpoint/checkpoint.py

```diff
@@ -76,6 +76,21 @@ def __init__(
             raise TypeError("A Checkpoint requires a valid DataContext")
         self._data_context = data_context
 
+        # if checkpoint was created without yml
+        if validations:
+            for val in validations:
+                if "batch_request" in val:
+                    if isinstance(val["batch_request"], RuntimeBatchRequest):
+                        batch_data = val["batch_request"]["runtime_parameters"][
+                            "batch_data"
+                        ]
+                        val["batch_request"] = val["batch_request"].to_json_dict()
```

Comment:
❤️ Very clever use of our own `BatchRequest.to_json_dict()`, which avoids the `DataFrame`. :-)

Reply:
... and then setting the `batch_data` to the actual `DataFrame`. :-)